jena-core/src/main/java/org/apache/jena/irix/IRIx.java

```diff
@@ -0,0 +1,194 @@
+import java.util.Objects;
+
+/**
+ * Support for RFC3986 IRIs.
+ * <p>
+ * The class {@code IRIx} is an abstraction layer; a provider is need to give an
+ * implementation. A provider is selected at start-up and is not expected to changes
+ * while the system is running.
+ * <p>
+ * Use {@link IRIs#reference} to check a string is suitable for us in RDF.
+ *
+ * @see IRIxResolver for resolving relative IRIs against a base.
+ * @see IRIs for functions related to IRIs
+ */
+public abstract class IRIx {
+
+    private final String iriString;
+
+    /**
+     * Create an {@link IRIx} without resolving the iri. This operation may throw an
+     * {@link IRIException}.
+     * <p>
+     * See {@link IRIs#resolve(String)} to create an absolute IRI, resolving
+     * against the system base if necessary.
+     * <p>
+     * See {@link IRIs#check(String)} to check a string is an absolute URI and is suitable for use in RDF.
+     * <p>
+     * See {@link IRIs#reference(String)} when the string is an absolute URI and
+     * should not be resolved against local system base (e.g. it was passed in from
+     * outside)to create an {@link IRIx} that is suitable for use in RDF.
+     */
+    static public IRIx create(String iri) throws IRIException {
+        Objects.requireNonNull(iri);
+        return SystemIRIx.getProvider().create(iri);
+    }
+
+    /**
+     * String must have been validated (e.g complies with the grammar of RFC3986).
+     * This constructor does not perform any additional checking.
+     */
+    protected IRIx(String string) {
+        this.iriString = string;
+    }
+
+    /**
+     * An <a href="https://tools.ietf.org/html/rfc3986#section-4.3"><em>absolute URI</em></a>
+     * is one with a URI scheme and without a fragment.
+     * The other components, host (authority), path, and query, are optional.
+     * <p>
+     * absolute-URI  = scheme ":" hier-part [ "?" query ]
+     * <p>
+     * Beware of the meaning : {@code http:abc} is an absolute URI - it has only a schema and a path without a root.
+     * <p>
+     * Note that a URI can be both "not absolute" and "not relative", e.g. {@code http://example/path#fragment}.
+     * <p>
+     * See {@linkplain #isReference()} for testing whether a URI is suitable for use in RDF.
+     */
+    public abstract boolean isAbsolute();
+
+    /**
+     * A <a href="https://tools.ietf.org/html/rfc3986#section-4.2"><em>relative
+     * URI</em></a> one without a scheme, and maybe without some of the other parts.
+     * <p>
+     * Often it is just the path part.
+     * <p>
+     * <p>
+     * See {@linkplain #isReference()} for testing whether a URI is suitable for use in RDF.
+     * <p>
+     * Note that a URI can be both "not absolute" and "not relative", e.g. {@code http://example/path#fragment}.
+     */
+    public abstract boolean isRelative();
+
+    /**
+     * An <em>RDF Reference</em> is an URI which has scheme.
+     * If it is hierarchical, it should a non-empty host- authority,
```

Comment:
s/it should a non-empty host- authority/it should be a non-empty host-authority? (although I think the `-` is not used elsewhere, so just `host authority`?)